File: JavaScript/0012.js
/**
 * @param {number} num
 * @return {string}
 */
var intToRoman = function(num) {
    // Define the values and symbols
    const values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1];
    const symbols = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"];
    
    let result = "";

    // Iterate over the values and symbols
    for (let i = 0; i < values.length; i++) {
        // Determine how many times the current value fits into num
        while (num >= values[i]) {
            // Append the corresponding symbol
            result += symbols[i];
            // Reduce num by the value
            num -= values[i];
        }
    }

    return result;
};

File: Go/0012.go
package main

import "fmt"

func intToRoman(num int) string {
	// Define the symbols and their corresponding values
	vals := []int{1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1}
	syms := []string{"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"}

	result := ""

	// Iterate over each value-symbol pair
	for i := 0; i < len(vals); i++ {
		// Determine how many times the current value fits into num
		count := num / vals[i]
		// Append the symbol count times
		for count > 0 {
			result += syms[i]
			count--
		}
		// Reduce num by the total value added
		num %= vals[i]
	}

	return result
}
